griptape/drivers/web_search/exa_web_search_driver.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from attrs import define, field
+
+from griptape.artifacts import JsonArtifact, ListArtifact
+from griptape.drivers import BaseWebSearchDriver
+from griptape.utils import import_optional_dependency
+from griptape.utils.decorators import lazy_property
+
+if TYPE_CHECKING:
+    from exa_py.api import Exa
+
+
+@define
+class ExaWebSearchDriver(BaseWebSearchDriver):
+    api_key: str = field(kw_only=True)
+    highlights: bool = field(default=False, kw_only=True)
+    use_auto_prompt: bool = field(default=False, kw_only=True)
+    params: dict[str, Any] = field(factory=dict, kw_only=True, metadata={"serializable": True})
+    _client: Exa = field(default=None, kw_only=True, alias="client")
+
+    @lazy_property()
+    def client(self) -> Exa:
+        return import_optional_dependency("exa_py").Exa(api_key=self.api_key)
+
+    def search(self, query: str, **kwargs) -> ListArtifact:
+        response = self.client.search_and_contents(
+            query=query,
+            num_results=self.results_count,
+            text=True,
+            **self.params,
+            **kwargs,
+        )
+        results = [
+            {"title": result.title, "url": result.url, "highlights": result.highlights, "text": result.text}
+            for result in response.results
+        ]
+        return ListArtifact([JsonArtifact(result) for result in results])
```

Comment:
Not for this, but do we have a conversation going to standardizing the web_search driver output? ListArtifact[BaseArtifact], [JsonArtifact], [TextArtifact}. Possibilities are endless (If you consider 3! endless)

Reply:
They should probably all be updated to `ListArtifact[JsonArtifact]`. Separate PR though.